app/assets/javascripts/preferences/panes/Security.tsx

```diff
@@ -1,9 +1,10 @@
 import { FunctionComponent } from 'preact';
 import { PreferencesPane } from '../components';
 import { TwoFactorAuthWrapper } from './two-factor-auth';
+import { MfaProps } from './two-factor-auth/MfaProps';
 
-export const Security: FunctionComponent = () => (
+export const Security: FunctionComponent<MfaProps> = (props) => (
   <PreferencesPane>
-    <TwoFactorAuthWrapper />
+    <TwoFactorAuthWrapper {...props} />
```

Comment:
Since props spreading is destructuring the props the children will only use the props they subscribe to. This code is equivalent to:
```
// Declaring and Initializing to better illustrate
interface PreferencesProps {
  closePreferences:()=>void;
  application:WebApplication;
  mfaGateway:MfaGateway;
}
const prefProps = {...randomPrefProps};

// this is the line equivalent to passing the props to the component.
const { mfaGateway } = prefProps;
```

So basically the child only acquires the props it needs.

Reply:
. If the `Security` component has X props and you use props

Reply:
> So basically the child only acquires the props it needs.

This isn't really true. If the Security component has X props and you use props spreading to pass them down to TwoFactorAuthWrapper, TwoFactorAuthWrapper will get all X props whether it uses them or not. Per React Docs:

> Spread attributes can be useful but they also make it easy to pass unnecessary props to components that don’t care about them or to pass invalid HTML attributes to the DOM. We recommend using this syntax sparingly.

Reply:
Ah, yeah, that's true. I conflated the spread with the destructuring.

Reply:
Fixed!

Reply:
Awesome! 🙂 can we do the same for props spreading on `PreferencesCanvas` and `PaneSelector`?

Reply:
Those components are gonna need everything but `closePreferences` so I'd leave them as they are. They're more of pass-through layout components which are gonna be responsible to handing over the appropriate props to the chidren.